ddtrace/appsec/_trace_utils.py

```diff
@@ -149,13 +152,17 @@ def track_user_login_failure_event(
     span = _track_user_login_common(tracer, False, metadata, login_events_mode)
     if not span:
         return
-
     if user_id:
         span.set_tag_str("%s.failure.%s" % (APPSEC.USER_LOGIN_EVENT_PREFIX_PUBLIC, user.ID), str(user_id))
-
     if exists is not None:
         exists_str = "true" if exists else "false"
         span.set_tag_str("%s.failure.%s" % (APPSEC.USER_LOGIN_EVENT_PREFIX_PUBLIC, user.EXISTS), exists_str)
+    if login:
+        span.set_tag_str("%s.failure.login" % APPSEC.USER_LOGIN_EVENT_PREFIX_PUBLIC, login)
+    if email:
+        span.set_tag_str("%s.failure.email" % APPSEC.USER_LOGIN_EVENT_PREFIX_PUBLIC, email)
```

Comment:
Do we not need to redact the email? is it a backend task?

Reply:
It's only for self instrumentation right now, so only if the customer wants to use it explicitly. And we already do it for successful login, this adds the support for failed login.